apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx

```diff
@@ -93,9 +91,9 @@ export const ConfirmTxScreen = () => {
                 <ConfirmTx
                   wallet={wallet}
                   unsignedTx={unsignedTx}
-                  onSuccess={() => {
+                  onSuccess={(signedTx) => {
                     closeModal()
-                    navigate.submittedTx()
+                    navigate.submittedTx(signedTx.signedTx.id)
```

Comment:
if there is animations on closeModal leverage the InteractionManager.runafter.....